micro-rdk/src/common/robot.rs

```diff
@@ -276,22 +277,46 @@ impl LocalRobot {
         // TODO: When cfg's on expressions are valid, remove the outer scope.
         #[cfg(feature = "data")]
         {
-            // TODO(RSDK-8125): Support selection of the DataStore trait other than
-            // DefaultDataStore in a way that is configurable
-            match DataManager::<DefaultDataStore>::from_robot_and_config(&robot, config) {
-                Ok(Some(mut data_manager)) => {
-                    if let Some(task) = data_manager.get_sync_task(robot.start_time) {
-                        let _ = robot.data_manager_sync_task.insert(Box::new(task));
+            match firmware_mode {
+                // TODO(RSDK-8125): Support selection of a DataStore trait other than
+                // DefaultDataStore in a way that is configurable
+                FirmwareMode::Normal => {
+                    match DataManager::<DefaultDataStore>::from_robot_and_config(&robot, config) {
+                        Ok(None) => {}
+                        Ok(Some(mut data_manager)) => {
+                            log::info!("running normal data management");
```

Comment:
i see the re is a log here. I would rather just have a log saying "firmware is in normal mode" or "firmware will sleep between cloud data sync" or something like that

Reply:
I'll move this logging to when the `FirmwareMode` is first parsed